File: client/others/Barco.php
<?php
class Barco {
    // Vacio -> 00
    // Barco -> 01
    // S Mina     -> 11
    // S Mina 2.0 -> 12
    // S Botiquin -> 13
    // S Radar    -> 14
    protected $mapa;
    protected $x = 0;
    protected $y = 4;
    protected $vida = 10;

    function __construct(&$mapa) {
        $this->mapa = &$mapa;
    }

    public function Generar(): void {
        echo "generando barco";
        $this->mapa[$this->x][$this->y] = 1;
        echo "vida: $this->vida";
    }

    private function Colision(): void {
        echo "colision";
        if ($this->mapa[$this->x][$this->y] == 11) {
            $this->vida -= 5;
            echo "mina: damage 5point";
        }
        if ($this->mapa[$this->x][$this->y] == 12) {
            $this->vida -= 5;
            echo "mina 2.0: damage 10point";
        } 
    }

    private function ComprobarVida(): void {
        if ($this->vida < 1) {
            $this->mapa[$this->x][$this->y] = 0;
        }
        echo "vida $this->vida";
    }

    public function Avanzar(): void {
        if ($this->mapa[$this->x][$this->y] == 0) {
            $this->mapa[$this->x][$this->y] = 0; //vaciamos la posicion actual
            $this->x++;
            $this->mapa[$this->x][$this->y] = 1; //asignamos a la nueva posicion
        }else {
            $this->Colision();
        }
        $this->ComprobarVida();
    }

    public function DesplazarDerecha(): void {
        if ($this->mapa[$this->x][$this->y] == 0) {
            $this->mapa[$this->x][$this->y] = 0; //vaciamos la posicion actual
            $this->y++;
            $this->mapa[$this->x][$this->y] = 1; //asignamos a la nueva posicion
        }else {
            $this->Colision();
        }
        $this->ComprobarVida();
    }

    public function DesplazarIzquierda(): void {
        if ($this->mapa[$this->x][$this->y] == 0) {
            $this->mapa[$this->x][$this->y] = 0; //vaciamos la posicion actual
            $this->y--;
            $this->mapa[$this->x][$this->y] = 1; //asignamos a la nueva posicion
        }else {
            $this->Colision();
        }
        $this->ComprobarVida();
    }
}

?>

File: client/others/Barco.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package proyecto;

public class Barco {
    public int[][] mapa;
    public int vida = 10;
    public int x = 0;
    public int y = 4;

    public Barco(int[][] mapa) {
        this.mapa = mapa;
    }

    public void Generar() {
        System.out.println("generando barco");
        this.mapa[this.x][this.y] = 1;
        System.out.println("vida: " + this.vida);
    }
    
    private void Colision() {
        System.out.println("Colision");
        if (this.mapa[this.x][this.y] == 11){
            this.vida -= 5;
        }else{
            this.vida -= 10;
        }
        System.out.println("vida: " + this.vida);
    }
    
    private void ComprobarVida() {
        if (this.vida < 1) {
            this.mapa[this.x][this.y] = 0;
        }
    }
    
    public void Avanzar() {
        this.mapa[this.x][this.y] = 0;
//        System.out.println("barco eliminado");
        this.x++;
        if (this.mapa[this.x][this.y] != 0){
            this.Colision();
        }
        this.mapa[this.x][this.y] = 1;
//        System.out.println("barco movido");
        this.ComprobarVida();
    }
    
    public void DesplazarDerecha (){
        this.mapa[this.x][this.y] = 0;
//        System.out.println("barco eliminado");
        this.y++;
        if (this.mapa[this.x][this.y] != 0){
            this.Colision();
        }
        this.mapa[this.x][this.y] =1;
//        System.out.println("barco movido");
        this.ComprobarVida();
    }
    
    public void DesplazarIzquierda (){
        this.mapa[this.x][this.y] = 0;
//        System.out.println("barco eliminado");
        this.y--;
        if (this.mapa[this.x][this.y] != 0){
            this.Colision();
        }
        this.mapa[this.x][this.y] =1;
//        System.out.println("barco movido");
        this.ComprobarVida();
    }  
            
 
}
